src/player.cpp

```diff
@@ -4823,6 +4823,32 @@ dealt_damage_instance player::deal_damage(Creature* source, body_part bp, const
         }
     }
 
+    int sum_cover = 0, coverage = 0;
+    for( item &i : worn )
+    {
+        if( i.covers( bp ) )
+        {
+            coverage = i.get_coverage();
+            sum_cover += coverage;
+        }
+    }
+    const int infection_chance = dealt_dams.type_damage( DT_BASH ) + ( dealt_dams.type_damage( DT_CUT ) + dealt_dams.type_damage( DT_STAB ) ) * 4 * sum_cover / 100;
```

Comment:
This scales based on how covered the body part is, regardless of whether the coverage comes from filty clothing or not.  It would make a lot more sense if it only summed up coverage from filthy clothing.  That would also eliminate the need to iterate over the worn clothing gain.